lua/tardis/sh_menuoptions.lua

```diff
@@ -29,26 +29,107 @@ if CLIENT then
-			comboBox.OnSelect = function(panel,index,value,data)
+			interior_combobox.OnSelect = function(panel,index,value,data)
 				TARDIS:SetSetting("interior",data,true)
 				LocalPlayer():ChatPrint("TARDIS interior changed. Respawn or repair the TARDIS for changes to apply.")
 			end
-			panel:AddItem(comboBox)
+			panel:AddItem(interior_combobox)
 
+			local visual_gui_toggle = vgui.Create("DCheckBoxLabel")
+			visual_gui_toggle:SetText("Enable new visual GUI")
+			visual_gui_toggle:SetValue(TARDIS:GetSetting("visual_gui_enabled"))
+			function visual_gui_toggle:OnChange(val)
+				TARDIS:SetSetting("visual_gui_enabled", val)
+			end
+			panel:AddItem(visual_gui_toggle)
+
+			local visual_gui_big_toggle = vgui.Create("DCheckBoxLabel")
+			visual_gui_big_toggle:SetText("Visual GUI big popup")
+			visual_gui_big_toggle:SetValue(TARDIS:GetSetting("visual_gui_bigpopup"))
+			function visual_gui_big_toggle:OnChange(val)
+				TARDIS:SetSetting("visual_gui_bigpopup", val)
+			end
+			panel:AddItem(visual_gui_big_toggle)
+
+			local DLabel3 = vgui.Create( "DLabel" )
+			DLabel3:SetText("Number of visual GUI button rows on the screen:")
+			panel:AddItem(DLabel3)
+			local visual_gui_screen_numrows = vgui.Create("DNumSlider")
+			visual_gui_screen_numrows:SetMinMax(2, 10)
+			visual_gui_screen_numrows:SetDecimals(0)
+			visual_gui_screen_numrows:SetValue(TARDIS:GetSetting("visual_gui_screen_numrows"))
+			function visual_gui_screen_numrows:OnValueChanged(val)
+				val = math.floor(val)
+				visual_gui_screen_numrows:SetValue(val)
+				TARDIS:SetSetting("visual_gui_screen_numrows", val)
+			end
+			panel:AddItem(visual_gui_screen_numrows)
+
+			local DLabel4 = vgui.Create( "DLabel" )
+			DLabel4:SetText("Number of visual GUI button rows in the popup:")
+			panel:AddItem(DLabel4)
+			local visual_gui_popup_numrows = vgui.Create("DNumSlider")
+			visual_gui_popup_numrows:SetMinMax(2, 10)
+			visual_gui_popup_numrows:SetDecimals(0)
+			visual_gui_popup_numrows:SetValue(TARDIS:GetSetting("visual_gui_popup_numrows"))
+			function visual_gui_popup_numrows:OnValueChanged(val)
+				val = math.floor(val)
+				visual_gui_popup_numrows:SetValue(val)
+				TARDIS:SetSetting("visual_gui_popup_numrows", val)
+			end
+			panel:AddItem(visual_gui_popup_numrows)
+
+			local DLabel4 = vgui.Create( "DLabel" )
+			DLabel4:SetText("TARDIS Visual GUI Theme:")
+			panel:AddItem(DLabel4)
+
+			local visual_gui_theme = vgui.Create("DComboBox")
+			visual_gui_theme:SetText("Visual GUI Theme")
+			local themes = {}
+			local theme_basefolder = TARDIS.visualgui_theme_basefolder
+			local files, folders = file.Find(theme_basefolder.."*", "GAME", "nameasc")
+			for k,folder in pairs(folders) do
+				if file.Exists(theme_basefolder..folder.."/default_on.png", "GAME")
+					and file.Exists(theme_basefolder..folder.."/default_off.png", "GAME")
+					and file.Exists(theme_basefolder..folder.."/background.png", "GAME")
+					and file.Exists(theme_basefolder..folder.."/theme.json", "GAME")
```

Comment:
This files existence is not allowed as part of GMA workshop upload rules found here: https://github.com/Facepunch/gmad/blob/master/include/AddonWhiteList.h
![image](https://user-images.githubusercontent.com/2363642/108838560-b4610480-75cb-11eb-94ce-80cf84bdd5ac.png)

We should change this to work like interiors such that there are files lua/tardis/themes like default.lua that define a theme and call TARDIS:AddTheme (not sure if we want AddTheme vs AddDesktopTheme yet) at the bottom of the file. Happy to work on this item with you if you're not sure how to do this

Reply:
Currently, I've removed all themes except for default and .json file in it.
So it only checks for default_on, default_off and background
Later, themes will be added with lua files, like interiors are. They will be configurable.